### ISSUE TITLE
Ability to automatically change the status using conditions

### ISSUE THREAD

**Issue by romainpoirier** [verbb/formie#508]
**What are you trying to do?**
It would be a powerful feature add automatic rules to change status based on a possibly incomplete form submission.

For example, on a very large multi-pages forms, the status could change each time a page or a group of pages are filled before the final submit.

This would ease the administration of submission states. Furthermore, this would enable the ability to setup, for example, several emails notifications using the status condition.

Currently, the status are only setup automatically on submit, and can only be changed from the CP. They don't act like a special field, which could change based on content.

**What's your proposed solution?**
Add status conditions, like the ones are currently existing for the fields or email notifications.

**Comment by engram-design** [comment]
Just to be clear, it looks like you're proposing two things here?

- Being able to set the status on every page submission (for multi-page forms)
- Update the status based on conditions

**Comment by romainpoirier** [comment]
Yes you're right, thank you for this summary.
`Update the status based on conditions` would come first, and then` Being able to set the status on every page submission (for multi-page forms)` would be very useful afterby.

**Comment by engram-design** [comment]
Yeah, the second part might be tricky to manage without a more major change, where we'd need to add status to each page settings, so you could set a status on every page submission.

For the first part, we'd need to introduce the same conditional-handler. I'll see what we can do here.

**Comment by romainpoirier** [comment]
Example in action: where all (required) fields of a page are filled and the page is submitted, add an action (event) to update the status.

Maybe one easier solution if this one is tricky would be to let the ability to change the submission status based on page or fields conditions. These fields could be setup manually in the condition.

Or maybe we could add an hidden field on each page, which content is the status we want when completed, and update the form status based on the last page of the form current progression.

**Comment by engram-design** [comment]
The more I've thought about this, the more potential caveat's I'm seeing crop up.

For example, if we had a mechanism to set the status based on the value of fields, what happens when the user modifies that? The form starts off with a "Stage 1" status. The user fills in 5 fields over 3 pages with specific values (according to the proposed conditions rules you'd create), then the status is changed to "Stage 2".

If we also plan to allow statuses to trigger emails, then that poses an issue when the user might navigate back to a page and change some values, turning the status back to "Stage 1".

I'll put this one on the back-burner for some more thought.

**Comment by romainpoirier** [comment]
In the case of this special multi-steps forms, I was thinking about a way to disable the go back button. So once a step is completed, it's completed: no way to re-trigger it, no multiple notification.

**Comment by idawgik** [comment]
I have a project that could really use some similar functionality. In my case, I would like to have the status on a submission set based on the value of a field. Don't need anything auto set based on pages of a form or anything, just if a dropdown value is x set status to y kinda thing.

**Comment by wmdhosting** [comment]
Now when sumbission is incomplete even if it all ok, but  viistor didnt not hit last submit button.. 

only option to change from incomplete to complete is to go to DB and change manually.. 

Will be great to add option same as for status.. to change to Complete in admin..

**Comment by engram-design** [comment]
Updated in [4.0.0-beta.4](https://github.com/verbb/formie/releases/tag/4.0.0-beta.4).
